docs: reduce the use of admonitions (#3371)
zmkfirmware/zmk: docs/docs/development/contributing/documentation.md

The documentation is written with [Docusaurus](https://docusaurus.io/). The ZMK source code has all of the necessary Docusaurus dependencies included, but referencing their documentation can be helpful at times.

> Latest LTS version of node is used to build the website, available at <https://nodejs.org/en/download/>.

Reviewer: ```suggestion
The website is built using the latest LTS version of node, which is available at <https://nodejs.org/en/download/>.
```

3. Ensure the sources are formatted properly and linted
4. Create a Pull Request for review and inclusion into the ZMK sources

:::note
If you are working with the documentation from within VS Code+Docker please be aware the documentation will not be auto-generated when making changes while the server is running. You'll need to restart the server when saving changes to the documentation.
:::

:::note
You will need `Node.js` and `npm` installed to update the documentation. If you're using the ZMK dev container (Docker) the necessary dependencies are already installed. Otherwise, you must install these dependencies yourself. Since `Node.js` packages in Linux distributions tend to be outdated, it's recommended to install the current version from a repository like [NodeSource](https://github.com/nodesource/distributions) to avoid build errors.
:::

> 

Author: The first note is not applicable, or at least it doesn't happen to me. Using VS Code with `npm run start` in devcontainers, editing `.md` files either inside or outside of the container triggers hot reload. It's either been solved since then, or a problem in a more specific environment.

The second note, I think is too specific to be helpful. Surely anyone working with node projects would know to maintain their node environments. Also it's specifically targeting Linux users. The `.nvmrc` file indicates this node project wants the lts version.

For these reasons I removed both.

Reviewer 2: TBH ZMK was my first real encounter with node, and ran into the issue with Ubuntu LTS packaged version being too old before noticing this. (Now docusaurus 3 is old enough that even distro versions have caught up?)

It might be worth keeping a notice saying ZMK documentation requires [at least node 20.0](https://docusaurus.io/docs/installation#requirements) and to make sure to install a compatible version.

Author: Added a short note about using latest LTS, matching the version specified in `docs/.nvmrc`.
